src/cares_wrap.cc

```diff
@@ -1473,11 +1473,16 @@ void AfterGetAddrInfo(uv_getaddrinfo_t* req, int status, struct addrinfo* res) {
       return Just(true);
     };
 
-    if (add(true, verbatim).IsNothing())
+    if (add(true, verbatim).IsNothing()) {
+      uv_freeaddrinfo(res);
```

Comment:
Can we do this with RAII to avoid accidentally introducing these types of leaks altogether? Similar to e.g. this line:

https://github.com/nodejs/node/blob/ce1c1a6c95d2ed1d9c4c94863f0da7bbb19f5bf2/src/cares_wrap.cc#L1671

Reply:
Done ! `AfterGetAddrInfo` return directly when `IsNothing` is true, is it expected ?

Reply:
Yes – what should it do?

Reply:
It seems the callback of `dns.lookup` never be called.

Reply:
You'd end up continuing the program after an uncaught exception, which is just a bad idea generally speaking. https://nodejs.org/api/process.html#warning-using-uncaughtexception-correctly has some more details on why exactly. :)

Reply:
Thanks!